### ISSUE TITLE
smcpp.estimation_tools error following thinning

### ISSUE THREAD

**Issue by austinhpatton** [popgenmethods/smcpp#53]
Hi there,

I've successfully run smc++ under settings using a mutation rate of 7.79e-9. However, when changing _only_ the mutation rate to 1.53e-8, the program runs into an error after loading in the data, running the first E-step, and plotting the first iteration of the model. The thinning process then begins and I'm hit with the following error.

166202 smcpp.estimation_tools ERROR (array([[    1,    -1,     0,     0],
       [43311,    -1,     0,     0],
       [    1,    -1,     0,     0],
       [22046,    -1,     0,     0],
       [34642,    -1,     0,     0],
       [30717,    -1,     0,     0],
       [ 6304,    -1,     0,     0]], dtype=int32), 3178)

Below is the resultant traceback.
Traceback (most recent call last):
  File "/home/austinp/anaconda3/bin/smc++", line 11, in <module>
    load_entry_point('smcpp==1.10.0', 'console_scripts', 'smc++')()
  File "/home/austinp/anaconda3/lib/python3.6/site-packages/smcpp/frontend/console.py", line 29, in main
    cmds[args.command].main(args)
  File "/home/austinp/anaconda3/lib/python3.6/site-packages/smcpp/commands/estimate.py", line 66, in main
    analysis = Analysis(args.data, args)
  File "/home/austinp/anaconda3/lib/python3.6/site-packages/smcpp/analysis.py", line 415, in __init__
    self._perform_thinning(args.thinning)
  File "/home/austinp/anaconda3/lib/python3.6/site-packages/smcpp/analysis.py", line 159, in _perform_thinning
    new_data = estimation_tools.thin_dataset(self._data, thinning)
  File "/home/austinp/anaconda3/lib/python3.6/site-packages/smcpp/estimation_tools.py", line 72, in thin_dataset
    return list(p.map(_thin_helper, zip(dataset, thinning)))
  File "/home/austinp/anaconda3/lib/python3.6/concurrent/futures/_base.py", line 556, in result_iterator
    yield future.result()
  File "/home/austinp/anaconda3/lib/python3.6/concurrent/futures/_base.py", line 398, in result
    return self.__get_result()
  File "/home/austinp/anaconda3/lib/python3.6/concurrent/futures/_base.py", line 357, in __get_result
    raise self._exception
  File "/home/austinp/anaconda3/lib/python3.6/concurrent/futures/thread.py", line 55, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/austinp/anaconda3/lib/python3.6/site-packages/smcpp/estimation_tools.py", line 62, in _thin_helper
    thinned = _smcpp.thin_data(ds, th)
  File "smcpp/_smcpp.pyx", line 473, in smcpp._smcpp.thin_data

It's not intuitive to me why I might be able to run the program without error under one mutation rate, but under another hit this error. 

Any idea how I might resolve this?

Thanks so much,
Austin

**Comment by terhorst** [comment]
The thinning interval, if not specified by the user, used to be set based on genetic distance, which was estimated from the recombination rate, which (again if not specified by the user) defaulted to be equal to the mutation rate. This explains why changing the mutation rate would lead to different behavior. In the most recent updates I have reverted this behavior in favor of something simpler based on sample size only (which is what we were doing when the program was first released). What was the actual error it threw?

**Comment by austinhpatton** [comment]
Thanks for your response! I recently updated to version 1.11.0dev, and since this update and specifying the thinning interval, I've just now gotten the program to run with this (previously error prone) mutation rate. When I was getting the error however I was still using the version distributed via anaconda. (1.9 I think?). I believe I copied the error in full above - I don't remember seeing anything else. 

Thanks for the additional information though! I'll be sure to continue using the updated version.
